src/material/slider/slider.ts

```diff
@@ -557,8 +558,17 @@ export class MatSlider
     const trackStyle = this._trackActive.nativeElement.style;
     trackStyle.left = styles.left;
     trackStyle.right = styles.right;
-    trackStyle.transform = styles.transform;
     trackStyle.transformOrigin = styles.transformOrigin;
+
+    if (styles.transformOrigin !== trackStyle.transformOrigin) {
+      this._elementRef.nativeElement.classList.add('mat-mdc-slider-disable-track-animation');
```

Comment:
This part seems a little complicated to me. Does the issue happen when the user is dragging?

Reply:
This is kind of a gross problem so apologies for the long description

The issue happens when a user alternates clicks between the two thumbs. This causes the track to change which thumb it's animating from, which is whichever thumb is not moving. If the animation is on, there will be a flicker as this change is made. BUT we don't want to disable animations entirely because then the thumb animation will be disabled, too

Another layer of complexity is that the track "origin" changing needs to happen immediately to avoid the flicker, but the change in `transform` still needs to have an animation so it moves along with the thumb. This is why I'm using `onStable` so that once the track animation origin has been changed we can do the actual translateX animation

Reply:
This is firing quite frequently though. If I add a `console.log`, I can see it firing 20 times just on init and then once for each pixel I drag. Toggling the class so frequently can cause style recalculations which affect performance.

Reply:
You're right, code was bad. It still fires a lot initially, but I think it's firing for all of the sliders that are in the other tabs of the demo, too